### PR TITLE
update storage changelog

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-common/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## 12.0.0 (2021-06-08)
 
+### Other Changes and Improvements
+
+- Fixed a filename encoding issue.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/CHANGELOG.md:7]: Isn't this a bug fix?

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## 12.0.0 (2021-06-08)
 
+### Other Changes and Improvements
+
+- Added and updated some samples.
+- Fixed a read consistency issue.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:8]: Isn't this a bug fix? It doesn't matter at this point, but just asking to understand the scope of the issue.

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@
 
 - Renamed `ContentLength` in `FileItemDetails` to `FileSize`.
 
+### Other Changes and Improvements
+
+- Updated some samples.
+- Fixed a read consistency issue.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:12]: Why do we have this across all package CLs? Was the issue present everywhere?

File: sdk/storage/azure-storage-common/CHANGELOG.md (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/CHANGELOG.md:7]: I wouldn't say it's a bug. We hadn't supported unicode filename before this fix was checked in. It's like an improvement or enhancement.

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md (hunk 3 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/CHANGELOG.md:12]: Yes, blob, datalake and file share services.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:8]: It's more of a behavior change than a bug fix.

**Pull request merged**